src/components/strategies/create/overlapping/CreateOverlappingStrategyBudget.tsx

```diff
@@ -39,9 +40,14 @@ export const CreateOverlappingStrategyBudget: FC<Props> = (props) => {
   const budgetTooSmall = isOverlappingBudgetTooSmall(order0, order1);
   const buyBudgetId = useId();
   const sellBudgetId = useId();
+  const { user } = useWeb3();
 
-  const checkInsufficientBalance = (balance: string, order: OrderCreate) => {
-    if (new SafeDecimal(balance).lt(order.budget)) {
+  const checkInsufficientBalance = (
+    balance: string,
+    order: OrderCreate,
+    user?: string
```

Comment:
the useEffect needs to be triggered on user change, to make it clear that this a dependency required I added it as a param. These are not callbacks and the useEffect has eslint hints disabled making it very prone for errors otherwise